y/iterator.go

```diff
@@ -76,11 +78,37 @@ func (v *ValueStruct) EncodeTo(buf []byte) []byte {
 
 // Iterator is an interface for a basic iterator.
 type Iterator interface {
+	// Next returns the next entry with different key on the latest version.
+	// If old version is needed, call NextVersion.
 	Next()
+	// NextVersion set the current entry to an older version.
+	// The iterator must be valid to call this method.
+	// It returns true if there is an older version, returns false if there is no older version.
+	// The iterator is still valid and on the same key.
+	NextVersion() bool
```

Comment:
We can return a `VersionIterator` from `Value()`, and the `VersionIterator` can perform `SeekVersion`, `Next`, etc...

`Next` and `NextVersion` isn't type-safe enough, because the `Iterator` still return values without any hint on type definition, we may forget to update some code to use the correct version, and the compiler cannot check it. We have forgotten to call `SetReadTS` in unistore's `Seek` method, so I prefer to put more logic into type level and let the compiler check it.

Reply:
We can do it after removing the Version from the Key() method